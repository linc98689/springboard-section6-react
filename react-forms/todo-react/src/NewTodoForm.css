.newtask-btn{
    margin:1px 5px 5px 5px;
}

.newtask-input{
    display:block;
    width:90%;
    padding: 3px 10px;
    border: 1px solid rgb(3, 99, 154);
}
input:focus{
    outline: 2px solid rgb(3, 99, 154);
    outline-offset: 1px;
    border:1px solid white;
}
.newform-warning{
    margin:1px;
    padding:0;
    font-size:0.7em;
    color:rgb(246, 14, 14);
}