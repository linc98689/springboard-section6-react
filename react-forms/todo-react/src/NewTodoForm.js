import React, {useState, useRef} from "react";
import "./NewTodoForm.css";


const NewTodoForm = ({addTask})=>{
    const [formData, setFormData] = useState("");
    const refNewTaskWarning = useRef();

    const handleSubmit = (evt)=>{
        evt.preventDefault();
        if(formData==="")
        return;
        addTask(formData);
        setFormData("");
    }

    const handleChange = (evt)=>{
        const task = evt.target.value;
        refNewTaskWarning.current.style.opacity = (task===""? 1:0);
        setFormData(data=>task);
    };

    return(<form>
        <input className="newtask-input" placeholder="new task" value={formData}
        onChange={handleChange}/>
        <p className="edit-form-warning" 
                style={{opacity:0}} ref={refNewTaskWarning}>required</p>
        <button   type="submit" className="newtask-btn"
        onClick={handleSubmit} >Add</button>
    </form>);
};

export default NewTodoForm;