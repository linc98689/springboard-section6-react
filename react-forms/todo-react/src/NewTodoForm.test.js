import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import NewTodoForm from "./NewTodoForm";

// smoke test
test("should render", ()=>{
    render(<NewTodoForm />)
});

// snapshot test
test("should not change", ()=>{
    const {asFragment} = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
})
