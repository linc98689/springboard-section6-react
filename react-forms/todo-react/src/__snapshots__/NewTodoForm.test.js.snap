// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not change 1`] = `
<DocumentFragment>
  <form>
    <input
      class="newtask-input"
      placeholder="new task"
      value=""
    />
    <button
      class="newtask-btn"
      type="submit"
    >
      Add
    </button>
  </form>
</DocumentFragment>
`;
